### ISSUE TITLE
Hypothesis testing with Recurrence Matrices

### ISSUE THREAD

**Issue by fargolo** [JuliaDynamics/RecurrenceAnalysis.jl#154]
Hello,

I intend to calculate the probabilities associated with 
structures in Rec. Matrices (trapping time, vertical, diagonal).

[I  sketched a way ](https://discourse.julialang.org/t/trying-julia-for-analytic-combinatorics/103682
) of calculating the probabilities associated with the sequences as binary words.
This allows one to calculate, for instance, the probability of a sequence (e.g. vertical run, 'trapping time')
considering its section (column size with all entries a_i,j). 

How would you include the information of the lagged sections (other columns)?
Could you point me to any references?

**Comment by Datseris** [comment]
Hi, the question falls outside my expertise but maybe @pucicu has some idea.

**Comment by pucicu** [comment]
Yes, I was already thinking about it, but to be honest, I don’t completely understand the question. I also do not get the point how the provided link is related to the question.

**Comment by fargolo** [comment]
Hello, @pucicu .
I was just reading your paper: https://link.springer.com/article/10.1140/epjs/s11734-022-00739-8

**Comment by fargolo** [comment]
Hello, @pucicu .

I was just reading [your paper](https://link.springer.com/article/10.1140/epjs/s11734-022-00739-8) and the problem is related to [section 2.3](https://link.springer.com/article/10.1140/epjs/s11734-022-00739-8#Sec5)

_"A fundamental finding was elaborated by [[67](https://link.springer.com/article/10.1140/epjs/s11734-022-00739-8#ref-CR67)], who mathematically developed the connection between correlation sum C(m) and the RQA measures recurrence rate RR(m), determinism DET(m), and average diagonal line length L(m). (...) These considerations have been further elaborated by [[68](https://link.springer.com/article/10.1140/epjs/s11734-022-00739-8#ref-CR68), [69](https://link.springer.com/article/10.1140/epjs/s11734-022-00739-8#ref-CR69)], which have further derived the analytical expressions for several RQA measures for certain stochastic processes, fractional Gaussian noise, and correlated noise (first analytical solutions were already given in [[70](https://link.springer.com/article/10.1140/epjs/s11734-022-00739-8#ref-CR70)]). Analytical and asymptotical expressions for RQA measures of specific random processes are important for defining baselines for benchmarking and testing."_

I did not have the chance to check these references yet,
The method that I linked calculates probabilities for consecutive runs in random binary sequences.

One could calculate the probability of observing L<sub>max</sub> or L<sub>avg</sub> if the column was random.

**Comment by Datseris** [comment]
_(I am closing this issue because it is not related with the code base RecurrenceAnalysis.jl, but rather it is a scientific question; please do feel free to continue the discussion as much as you would like here!)_
